tests/dao_test/user_dao_test.rs

```diff
@@ -0,0 +1,77 @@
+use super::init_db_context;
+
+#[actix_rt::test]
+async fn add_user_returns_1() -> Result<(), sqlx::Error> {
+    let db = init_db_context().await;
+    let email = "gigi@email.com";
+    let password = "fake-password";
+
+    let result = db.users.add_user(email, password).await;
+
+    assert!(result.is_ok());
+    assert_eq!(1, result.unwrap());
+
+    let del_res = db.users.delete_user(email).await?;
+    assert_eq!(1, del_res);
+
+    Ok(())
```

Comment:
that's a good point. Can you suggest me the syntax to do so?

Reply:
by changing into this
```
async fn add_user<'a>(
    &self, 
    transaction: &Transaction<'a, Postgres>, 
    email: &str, 
    password: &str
) -> sqlx::Result<u64> {
    // Your existing logic here, using `transaction` instead of `db`.
}
```
means that a transaction struct must be always passed to the function. If it's not using a transaction to perform the query, the transaction argument should be an option; if empty, uses self, correct?

Reply:
db has type pool that can be passed to execute the query, but transaction does not have it. How can it execute the query?